common/src/commonMain/kotlin/com.harmony.kotlin/data/datasource/network/GenericNetworkQuery.kt

```diff
@@ -31,3 +31,49 @@ GetPasswordTokenInteractor) : PaginationOffsetLimitQuery(identifier ?: "$offset-
 class OAuthPasswordObjectQuery<T>(value: T, val getPasswordTokenInteractor: GetPasswordTokenInteractor) : ObjectQuery<T>(value),
     OAuthClientQuery
 
+abstract class GenericNetworkQuery(
+    val path: String,
+    val params: List<Pair<String, String>>?,
+    val headers: List<Pair<String, String>>?,
+    cacheKey: String? = null) : KeyQuery(cacheKey ?: path)
+
+abstract class GenericOAuthNetworkQuery(
+    val getPasswordTokenInteractor: GetPasswordTokenInteractor,
+    path: String,
+    params: List<Pair<String, String>>? = emptyList(),
+    headers: List<Pair<String, String>>? = emptyList(),
+    cacheKey: String? = null) : GenericNetworkQuery(path, params, headers, cacheKey)
+
+// PUT Create JavaDoc
+abstract class GenericIdNetworkQuery(
+    val id: Int,
+    path: String,
+    params: List<Pair<String, String>>? = emptyList(),
+    headers: List<Pair<String, String>>? = emptyList(),
+    cacheKey: String? = null) : GenericNetworkQuery(path, params, headers, cacheKey)
+
+abstract class GenericOAuthIdNetworkQuery(
```

Comment:
**GenericOAuthIdNetworkQuery** should extend from **GenericIdNetworkQuery** because it's the same thing just adding the password token interactor.

Reply:
If we extend from **GenericOAuthNetworkQuery** yes or yes the user is going to provide a getPasswordInteractor and we can do query.getpasswordtoken. 

If not, the user could forget the getPasswordInteractor and we cannot access to the getPasswordInteractor property

Reply:
We create a GenericOAuthInterface that fixes the issue.